applications/ParticleMechanicsApplication/custom_constitutive/hencky_plastic_3D_law.cpp

```diff
@@ -74,7 +74,15 @@ HenckyElasticPlastic3DLaw::~HenckyElasticPlastic3DLaw()
 
 bool HenckyElasticPlastic3DLaw::Has( const Variable<double>& rThisVariable )
 {
-    return false;
+    if (rThisVariable == MP_DELTA_PLASTIC_STRAIN
+        || rThisVariable == MP_EQUIVALENT_PLASTIC_STRAIN
+        || rThisVariable == MP_DELTA_PLASTIC_VOLUMETRIC_STRAIN
+        || rThisVariable == MP_ACCUMULATED_PLASTIC_VOLUMETRIC_STRAIN
+        || rThisVariable == MP_DELTA_PLASTIC_DEVIATORIC_STRAIN
+        || rThisVariable == MP_ACCUMULATED_PLASTIC_DEVIATORIC_STRAIN
+        || rThisVariable == MIU)
```

Comment:
I am not sure this should be put here. Can you put it in your law instead? Not all model has these I guess.

Reply:
They are here because they are being called by default (for EVERY law) in the updated lagrangian element finalize sln step. Now the element asks the law (which generally does not have these variables) if it has these variables, and gets them if it does. These additions actually increase safety.